From 96c3eac10c833f8654f98bb4ec8f1ba22cca5897 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 13 Feb 2023 15:30:55 +0800
Subject: [PATCH 02/20] HACK: media: gpu: v4l2: Allow V4L2 VEA on non-chromeos
 platform

NOTE(crbug.com/901264):
Encoders use hack for passing offset within a DMA-buf, which is not
supported upstream.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 media/gpu/gpu_video_encode_accelerator_factory.cc | 3 ++-
 media/gpu/v4l2/BUILD.gn                           | 2 +-
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/media/gpu/gpu_video_encode_accelerator_factory.cc b/media/gpu/gpu_video_encode_accelerator_factory.cc
index 71d15a3..800f72f 100644
--- a/media/gpu/gpu_video_encode_accelerator_factory.cc
+++ b/media/gpu/gpu_video_encode_accelerator_factory.cc
@@ -51,7 +51,8 @@ std::unique_ptr<VideoEncodeAccelerator> CreateV4L2VEA() {
   scoped_refptr<V4L2Device> device = V4L2Device::Create();
   if (!device)
     return nullptr;
-#if BUILDFLAG(IS_CHROMEOS)
+/* HACK: Enable V4L2 VEA for v4l-rkmpp */
+#if 1 // BUILDFLAG(IS_CHROMEOS)
   // TODO(crbug.com/901264): Encoders use hack for passing offset within
   // a DMA-buf, which is not supported upstream.
   return base::WrapUnique<VideoEncodeAccelerator>(
diff --git a/media/gpu/v4l2/BUILD.gn b/media/gpu/v4l2/BUILD.gn
index 9cc44f7..5f0caac 100644
--- a/media/gpu/v4l2/BUILD.gn
+++ b/media/gpu/v4l2/BUILD.gn
@@ -85,7 +85,7 @@ source_set("v4l2") {
     }
   }
 
-  if (is_chromeos) {
+  if (true) { # if (is_chromeos) {
     sources += [
       # TODO(crbug.com/901264): Encoders use hack for passing offset
       # within a DMA-buf, which is not supported upstream.
-- 
2.20.1

